acvm-repo/bn254_blackbox_solver/src/fixed_base_scalar_mul.rs

```diff
@@ -53,8 +53,21 @@ pub fn embedded_curve_add(
     input2_x: FieldElement,
     input2_y: FieldElement,
 ) -> Result<(FieldElement, FieldElement), BlackBoxResolutionError> {
-    let mut point1 = grumpkin::SWAffine::new(input1_x.into_repr(), input1_y.into_repr());
-    let point2 = grumpkin::SWAffine::new(input2_x.into_repr(), input2_y.into_repr());
+    fn create_point(
+        x: FieldElement,
+        y: FieldElement,
+    ) -> Result<grumpkin::SWAffine, BlackBoxResolutionError> {
+        match std::panic::catch_unwind(|| grumpkin::SWAffine::new(x.into_repr(), y.into_repr())) {
+            Ok(point) => Ok(point),
+            Err(_) => Err(BlackBoxResolutionError::Failed(
+                BlackBoxFunc::EmbeddedCurveAdd,
+                format!("Point ({}, {}) is not on curve", x.to_hex(), y.to_hex()),
+            )),
+        }
```

Comment:
```suggestion
        let point = grumpkin::SWAffine::new_unchecked(x.into_repr(), y.into_repr());
        if !point.is_on_curve() {
            return Err(BlackBoxResolutionError::Failed(
                BlackBoxFunc::EmbeddedCurveAdd,
                format!("Point ({}, {}) is not on curve", x.to_hex(), y.to_hex())
        };
        if !point.is_in_correct_subgroup_assuming_on_curve() {
            return Err(BlackBoxResolutionError::Failed(
                BlackBoxFunc::EmbeddedCurveAdd,
                format!("Point ({}, {}) is not in correct subgroup", x.to_hex(), y.to_hex())
        };
        point
```

We should avoid trying to catch unwinding panics if it's unnecessary. This way we can also give a more specific error message.

I think we should also just move `create_point` outside of this `embedded_curve_add`

Reply:
I had the same concerns regarding panics- I had elected to take this route since I didn't want to make assumptions about the underlying validation that ark-ec performs for fear that it might change out from under us and we would have no idea (unless they simultaneously bump the version to 0.5.0). I think the real fix is having a different constructor in that ark-ec that returns a `Result`. I might post a PR over there to do so.

In the meantime, I'll lift the validation as you suggested.

I had also thought similarly on lifting `create_point`: I had embedded it since this was the only place we call `SWAffine::new`. 

But sounds good, I'll lift that out as well.